microcosm_daemon/daemon.py

```diff
@@ -0,0 +1,133 @@
+"""
+Base class for command-line driven asynchronous worker.
+
+"""
+from abc import ABCMeta, abstractmethod, abstractproperty
+from argparse import ArgumentParser
+from logging import getLogger
+
+from microcosm.api import create_object_graph
+from microcosm.loaders import load_from_environ
+
+from microcosm_daemon.api import StateMachine
+from microcosm_daemon.runner import ProcessRunner, SimpleRunner
+
+
+class Daemon(object):
+    __metaclass__ = ABCMeta
+
+    @abstractproperty
+    def name(self):
```

Comment:
The intention is that subclasses must define `name` and `__call__` (below) to correctly initialize the object graph and define the actual worker function. (We'll have a subclass in the pubsub layer that implements a message consumer loop for `__call__` by default.)